api/config_server.go

```diff
@@ -41,8 +41,16 @@ func (s *server) Delete(_ context.Context, sel *pb.Selector) (*pb.DeleteRecordsR
 		return nil, err
 	}
 
+	ids := make([]string, 0, len(recs))
 	for _, r := range recs {
-		if err = s.config.delete(r.GetId()); err != nil {
+		ids = append(ids, r.GetId())
+	}
+	if _, err = s.disconnectLocked(ids); err != nil {
```

Comment:
Does disconnect return an error if a connection isn't already connected?

https://github.com/pomerium/cli/blob/ae250e91e9b616e2675ad0312f8fd43850b581f8/api/listener.go#L43

Reply:
the only error kind that may be returned is that there's no active listener. 
the `disconnectLocked` would fill in `ListenerStatus.LastError` field, which we ignore during record deletion.